### PR TITLE
Allow passing sorting key in fitsheader

### PR DESCRIPTION
**Pull request by krachyon**
### Description

With `store_true` for the `-f` argument to `fitsheader`, it is not possible to pass `SORT_KEYWORD` as the additional argument, even though it's mentioned in the help.  This PR corrects the functionality and a test-case verifying that it works as desired.

The sorting functionality was already present in the code, so I assume this is how it was supposed to work. It will however now break if you pass `-f` as the last argument before the filename `fitsheader -k KEY -f foo.fits` as the file is interpreted as an argument to `-f`. If someone knows how to force the `--fitsort=KEY` syntax for this argument it may be an improvement.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [ ] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [ ] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by pllim** [comment]
Thanks!

Not sure about What's New... probably not unless subpackage maintainer thinks otherwise.

Also probably need to add to user facing documentation.

File: astropy/io/fits/tests/test_fitsheader.py [hunk 1]
```diff
@@ -121,13 +121,33 @@ def test_fitsort(self, capsys):
         assert out[3].endswith('test0.fits 19/05/94')
 
         # check that COMMENT and HISTORY are excluded
-        fitsheader.main(['-e', '0', '-f', self.data('tb.fits')])
+        fitsheader.main(['-f', '-e', '0', self.data('tb.fits')])
```
Comment (pllim) [inline astropy/io/fits/tests/test_fitsheader.py:124]: Why this change?

File: docs/changes/io.fits/13011.bugfix.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Allow passing ``SORT_KEYWORD`` as an argument to ``fitsheader``:
```
Comment (pllim) [inline docs/changes/io.fits/13011.bugfix.rst:1]: I don't think this is a bug fix. Please change it to "api". Thanks!

**Comment by krachyon** [comment]
> Not sure about What's New... probably not unless subpackage maintainer thinks otherwise.
Added it anyway to appease the bot  

> Also probably need to add to user facing documentation.
It shows the following when calling `fitsheader --help` now:
``` 
-f [SORT_KEYWORD], --fitsort [SORT_KEYWORD]
                        print the headers as a table with each unique keyword in a given column (fitsort format); if a SORT_KEYWORD is specified, the result will be sorted along that keyword
```
is that sufficient?

File: astropy/io/fits/tests/test_fitsheader.py (hunk 1 above)
Comment (krachyon) [reply astropy/io/fits/tests/test_fitsheader.py:124]: It'll interpret the filename as the argument to `-f` otherwise. I'm not sure if there's a way around this with argparse, apart from a separate `sort_key` argument...

Comment (pllim) [reply astropy/io/fits/tests/test_fitsheader.py:124]: Does sound like a breaking change. Let's see what subpackage maintainer(s) think.

**Comment by saimn** [comment]
> The sorting functionality was already present in the code, so I assume this is how it was supposed to work. It will however now break if you pass -f as the last argument before the filename fitsheader -k KEY -f foo.fits as the file is interpreted as an argument to -f

Actually this sorting feature was removed during the development of the PR (#7240), because of this issue with `-f file.fits` : https://github.com/astropy/astropy/pull/7240/commits/893d0d9274ee06bffec4fabc21454e4fffafc43f
Since we want to keep the possibility to run `fitsheader -f *.fits` and the sorting feature is already available with `-k`, I think we should keep things like that and what's need to be fixed is just the help (the "if a SORT_KEYWORD is specified..." sentence should be removed).

**Comment by krachyon** [comment]
> Actually this sorting feature was removed during the development of the PR (#7240), because of this issue with `-f file.fits` : [893d0d9](https://github.com/astropy/astropy/commit/893d0d9274ee06bffec4fabc21454e4fffafc43f) Since we want to keep the possibility to run `fitsheader -f *.fits` and the sorting feature is already available with `-k`, I think we should keep things like that and what's need to be fixed is just the help (the "if a SORT_KEYWORD is specified..." sentence should be removed).

The behaviour I implemented is also documented as such (https://docs.astropy.org/en/stable/io/fits/usage/scripts.html?highlight=fitsheader#module-astropy.io.fits.scripts.fitsheader):
```
9. Same as above, sorting the output along a specified keyword::

$ fitsheader -f DATE-OBS -e 0 -k DATE-OBS -k ESO.INS.ID *.fits
```

From both the documentation and the code I don't really see how sorting with just `-k` is supposed to work. If I do
`fitsort -k OBJECT *.fits` i get the order in which the shell resolves the wildcard.

Considering that sorting is a bit of an orthogonal task to both selecting keywords with `-k` and changing the format with `-f`, what about making the sorting keyword it's own argument? Then it could work e.g. like `fitsort --sort=OBJECT` no matter if any `-k` or `-f`arguments are specified. It could also be `nargs=1` so no chance of the user getting confused by passing it as the last argument and picking up a filename.

**Comment by saimn** [comment]
Oh sorry, `-k` doesn't sort the columns, it just selects them. So yes I guess the best solution would be to add a new keyword to allow sorting the columns. We could even allow using this keyword multiple times to sort on multiple columns.

**Comment by krachyon** [comment]
> Oh sorry, `-k` doesn't sort the columns, it just selects them. So yes I guess the best solution would be to add a new keyword to allow sorting the columns. We could even allow using this keyword multiple times to sort on multiple columns.

Should I implement this here or close this and open another PR?
